api.js

```diff
@@ -4,76 +4,88 @@ require('colors');
 
 const { baseLinks } = require('./commands/link');
 const utils = require('./utils/utils');
+const { parseFileResponse, convertToFileType, file } = require('./utils/file-utils');
 const console = require('./utils/console');
 const invoke = require('./lib/invoke');
 
 const linksPath = path.join(utils.sharedDirectoryPath(), 'links.json');
 
-module.exports.run = (action, d, cb) => {
-  if (this.key && this.key.startsWith('demo')) {
-    return maybe(cb, new Promise((resolve) => {
-      console.log('This is a demo API key!'.red);
-      console.log('To get your own, run:');
-      console.log('');
-      console.log(`  ${'$'.grey} ${'api signup'.yellow}`);
-      console.log('');
+const scope = {};
 
-      return resolve();
-    }));
+module.exports.config = (apiKey) => {
+  scope.key = apiKey;
+
+  function setService(service) {
+    scope.service = service;
+    return api;
   }
+  return Object.assign(setService.bind(scope), api);
+};
 
-  // Can't reassign params
-  let data = d;
-  let callback = cb;
+const api = {
+  run(action, d, cb) {
+    if (scope.key && scope.key.startsWith('demo')) {
+      return maybe(cb, new Promise((resolve) => {
+        console.log('This is a demo API key!'.red);
+        console.log('To get your own, run:');
+        console.log('');
+        console.log(`  ${'$'.grey} ${'api signup'.yellow}`);
+        console.log('');
 
-  // If no data is passed in, default to {}
-  if (typeof data === 'function') {
-    callback = data;
-    data = {};
-  } else if (!data) {
-    data = {};
-  }
+        return resolve();
+      }));
+    }
 
-  // Don't call api if there is a local link
-  const localLinks = utils.fileExists(linksPath) ? require(linksPath) : baseLinks;
-  if (localLinks.services[this.service] && localLinks.directories[process.cwd()].length) {
-    const handler = require(path.join(process.cwd(), 'node_modules/api/utils/handler.js'));
-    const errors = utils.buildErrors(localLinks.services[this.service]);
+    // Can't reassign params
+    let data = d;
+    let callback = cb;
 
-    const event = {
-      name: action,
-      data,
-      pathOverride: localLinks.services[this.service],
-      errors,
-    };
+    // If no data is passed in, default to {}
+    if (typeof data === 'function') {
+      callback = data;
```

Comment:
It looks as though we've lost some coverage here... did this used to be covered?

Reply:
I'm pretty sure it never was